erc/ERCS/erc-4337.md

```diff
@@ -434,6 +434,16 @@ Otherwise, attackers may be able to use the banned opcodes to detect running on-
 
 When a bundler includes a bundle in a block it must ensure that earlier transactions in the block don't make any UserOperation fail. It should either use access lists to prevent conflicts, or place the bundle as the first transaction in the block.
 
+
+## Rationale
+
+The main challenge with a purely smart contract wallet based account abstraction system is DoS safety: how can a block builder including an operation make sure that it will actually pay fees, without having to first execute the entire operation? Requiring the block builder to execute the entire operation opens a DoS attack vector, as an attacker could easily send many operations that pretend to pay a fee but then revert at the last moment after a long execution. Similarly, to prevent attackers from cheaply clogging the mempool, nodes in the P2P network need to check if an operation will pay a fee before they are willing to forward it.
```

Comment:
`smart contract wallet based account abstraction system` is a mouthful.

"The main challenge of implementing an Account Abstraction system that is based purely on Smart Contract Wallets, lies in designing protections against potential DoS attack vectors."
Something like that?

Reply:
I think that the start of a rationale section should have a short sentence summarizing the actual rationale, not going about it indirectly with the main challenge first.